### PR TITLE
plat: xilinx: code restructure

### PR DESCRIPTION
**Pull request by jollysxilinx**
This patchset restructures code to move common code to separate directory

Signed-off-by: Jolly Shah jollys@xilinx.com

### CODE REVIEW

File: plat/xilinx/zynqmp/zynqmp_ipi.c [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2017, ARM Limited and Contributors. All rights reserved.
+ * Copyright (c) 2017-18, ARM Limited and Contributors. All rights reserved.
```
Comment (antonio-nino-diaz-arm) [inline plat/xilinx/zynqmp/zynqmp_ipi.c]: Write the full year.

File: plat/xilinx/zynqmp/include/zynqmp_def.h [hunk 2]
```diff
@@ -7,7 +7,7 @@
 #ifndef ZYNQMP_DEF_H
 #define ZYNQMP_DEF_H
 
-#include <plat/common/common_def.h>
+#include <common_def.h>
```
Comment (antonio-nino-diaz-arm) [inline plat/xilinx/zynqmp/include/zynqmp_def.h]: Using the file name is deprecated, now includes have to use the full path.

File: plat/xilinx/zynqmp/include/zynqmp_def.h [hunk 3]
```diff
@@ -340,4 +347,4 @@
 #define  AFIFM6_WRCTRL		U(13)
 #define  FABRIC_WIDTH		U(3)
 
-#endif /* ZYNQMP_DEF_H */
+#endif /* __ZYNQMP_DEF_H__ */
```
Comment (antonio-nino-diaz-arm) [inline plat/xilinx/zynqmp/include/zynqmp_def.h]: This change is incorrect.

File: plat/xilinx/zynqmp/include/plat_ipi.h [hunk 4]
```diff
@@ -1,13 +1,13 @@
 /*
- * Copyright (c) 2017, ARM Limited and Contributors. All rights reserved.
+ * Copyright (c) 2017-18, ARM Limited and Contributors. All rights reserved.
```
Comment (antonio-nino-diaz-arm) [inline plat/xilinx/zynqmp/include/plat_ipi.h]: Full year.

File: plat/xilinx/common/ipi.c [hunk 5]
```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2017-18, ARM Limited and Contributors. All rights reserved.
```
Comment (antonio-nino-diaz-arm) [inline plat/xilinx/common/ipi.c]: Full years.

File: plat/xilinx/common/ipi.c [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/*
+ * Xilinx IPI agent registers access management
+ */
+
+#include <errno.h>
+#include <ipi.h>
```
Comment (antonio-nino-diaz-arm) [inline plat/xilinx/common/ipi.c]: First C includes, then TF includes, then platform includes:

```
#include <errno.h>
#include <string.h>

#include <common/debug.h>
#include <common/runtime_svc.h>
#include <lib/bakery_lock.h>
#include <lib/mmio.h>

#include <ipi.h>
#include <plat_ipi.h>
#include <plat_private.h>
```

Same for the other files.

File: plat/xilinx/common/ipi.c (hunk 5 above)
Comment (antonio-nino-diaz-arm) [inline plat/xilinx/common/ipi.c]: Full year.

**Comment by jollysxilinx** [comment]
Thanks for the review. Fixed all comments and force pushed changes on same branch.

**Comment by antonio-nino-diaz-arm** [comment]
jenkins: test this please

**Comment by antonio-nino-diaz-arm** [comment]
The failures are just a few MISRA defects.

**Pull request merged**